docs/developer-guide/cstor-pool-docs.md

```diff
@@ -0,0 +1,189 @@
+
+### CSPC-Operator Introduction
+
+This sections describes the custom resources and components involved in cStor pools.
+
+```
+                                                              +-------------------------+
+                                                              |                         |
+                                                              |                         |
+                                                              |                         |
+                                                              |                         |
+                                                              |      CSPC-Operator      |
+                                                              |                         |
+                               XXXXXXXXXXXXXXX                |                         |
+                              X              X                |                         |
+Admin/User/SRE               X               X                |                         |
+                            X                X  Reconcile     |                         |
+     XXX                   X                 +--------------->+                         |
+    XXXXX                  X                 X                |                         |
+      X                    X   CSPC-YAML     X                |                         |
+    XXXXX                  X                 X                |                         |
+      X        Applies     X                 X                +-------------------------+
+      X     +------------->X                 X
+     XXX                   X                 X
+    X X X                  X                 X
+      X                    X                 X
+                           X                 X
+                           XXXXXXXXXXXXXXXXXXX
+
+```
+
+1. CSPC contains a list of pool specs and a pool spec contains configuration for cStor pool creation on a particular node as specified in the pool spec.
+For example, if a CSPC manifest has 3 pool specs, 3 cStor pools will be created on 3 different nodes.
+
+2. An admin/SRE can preapre their CSPC YAML according to the reqiremnts and do a `kubectl apply -f <cspc-yaml-file>` to provision a CStorPoolCluster.
+
+3. Once a CSPC is applied, it is validated by cStor admission server and if the validation passes it is admitted to the etcd.
+
+4. The `CSPC-operator` receives events from Kubernetes APIs  for the created CSPC as well as existing CSPC and `CSPC-Operator` converges the system towards intent specified in the CSPC. Just a note that `CSPC-Operator` is deployed as a deployment controller and runs as a pod in the Kubernetes cluster.
+
+### What does CSPC-Operator do?
+
+1. CSPC Operator creates the following resources as part of its job to converge the system towards the intent specified in the CSPC:
+- CStorPoolInstance(CSPI) - CSPI is a Kubernetes custom resource like the CSPC.
```

Comment:
Should we need to say CSPI will belong to a particular node?

Reply:
I feel it is too early to say it here. I have clarified this later in the doc.